aten/src/ATen/core/type.cpp

```diff
@@ -870,14 +870,29 @@ VaryingShape<Stride> TensorType::computeStrideProps(
   std::vector<size_t> stride_indices(sizes.size());
   std::iota(stride_indices.begin(), stride_indices.end(), 0);
 
+  // Sorting strides in ascending order
+  // Warning: With more than 2 strides of the same size
+  // there is insufficient information to recreate the contiguous order
+  // Ties are broken based on whether one of the dimensions is of size
+  // one.  When two dimensions have the same stride, the stride 
+  // associated with a dimension of size 1 is considered "smaller"
+  // as it created the condition for the second stride of the same size
+  // Example:
+  //  Prior to sorting
+  //  Idx:     [0,   1,  2,  3]
+  //  sizes:   [8,   1, 10, 16]
+  //  Strides: [160, 1, 16,  1]
+  //  After sorting
+  //  Idx:     [1,  3,  2,   0]
+  //  sizes:   [1, 16, 10,   8]
+  //  Strides: [1,  1, 16, 160]
+
   std::sort(
       stride_indices.begin(),
       stride_indices.end(),
-      [&strides](const int& a, const int& b) {
-        // break ties in case of unsqueezed dims
-        // i.e. (1, 1, 5)
+      [&strides, &sizes](const int& a, const int& b) {
         if (strides[a] == strides[b]) {
-          return a > b;
+          return sizes[a] == 1;
```

Comment:
nitpick: let's play on the safe side. In the ambiguous case, let's default NCHW.

i.e. `sizes[a] == 1 if sizes[b] != sizes[a] else a > b`